grdwindinversion/gradientFeatures.py

```diff
@@ -0,0 +1,448 @@
+        """
+        Convert wind direction to meteorological convention by creating a new 'angle' DataArray.
+
+        Parameters
+        ----------
+        streaks : xarray.Dataset
+            Dataset containing the streaks.
+
+        Returns
+        -------
+        xarray.Dataset
+            Dataset containing the streaks with wind direction in meteorological convention.
+
+        """
+        streaks_meteo = self.xr_dataset[['longitude', 'latitude', 'ground_heading', 'ancillary_wind']].interp(
+            line=streaks.line,
+            sample=streaks.sample,
+            method='nearest')
+
+        streaks_meteo['angle'] = xsarsea.dir_sample_to_meteo(
+            np.rad2deg(streaks['angle']), streaks_meteo['ground_heading'])
+        streaks_meteo['angle'].attrs[
+            'winddir_convention'] = "Wind direction in meteorological convention (clockwise, from), ex: 0°=from north, 90°=from east"
+
+        return streaks_meteo
+
+    def streaks_smooth_mean(self):
+        """
+        Compute streaks by smoothing the histograms first and then computing the mean.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        xarray.DataArray
+            DataArray containing the streaks.           
+        """
+
+        try:
+            hist_smooth = self.hist.copy()
+            hist_smooth['weight'] = xsarsea.gradients.circ_smooth(
+                hist_smooth['weight'])
+
+            # Compute the mean across 'downscale_factor', 'window_size', and 'pol'
+            hist_smooth_mean = hist_smooth.mean(
+                ['downscale_factor', 'window_size', 'pol'])
+
+            # Select histogram peak
+            iangle_smooth_mean = hist_smooth_mean['weight'].fillna(
+                0).argmax(dim='angles')
+            streaks_dir_smooth_mean = hist_smooth_mean['angles'].isel(
+                angles=iangle_smooth_mean)
+            streaks_weight_smooth_mean = hist_smooth_mean['weight'].isel(
+                angles=iangle_smooth_mean)
+
+            # Combine angles and weights into a dataset
+            streaks_smooth_mean = xr.Dataset({
+                'angle': streaks_dir_smooth_mean,
+                'weight': streaks_weight_smooth_mean
+            })
+
+            # Remove 'angles' coordinate
+            streaks_smooth_mean = streaks_smooth_mean.reset_coords(
+                'angles', drop=True)
+
+            # Remove ambiguity with ancillary wind
+            streaks_smooth_mean = self._remove_ambiguity(
+                streaks_smooth_mean)
+
+            # Convert to meteo convention
+            streaks_smooth_mean = self.convert_to_meteo_convention(
+                streaks_smooth_mean)
+
+            # Set attributes
+            streaks_smooth_mean['angle'].attrs['description'] = 'Wind direction estimated from local gradient; histograms smoothed first, then mean computed'
+
+            return streaks_smooth_mean
+
+        except Exception as e:
```

Comment:
This try except is huge, could it be reduced to critical part with the proper exception raised (eg ValueError, IOError,...)

Reply:
#91